### PR TITLE
Add Free Willy Keyboard

### PR DESCRIPTION
**Pull request by Tsquash**
## Description
Adds the free willy keyboard definitions.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/free_willy/rules.mk [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (drashna) [inline keyboards/free_willy/rules.mk:1]: This needs MCU and basic configuration settings

Comment (Tsquash) [reply keyboards/free_willy/rules.mk]: done, thank you!

**Review by drashna — APPROVED** [review]

File: keyboards/free_willy/config.h [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+#define UNUSED_PINS
+
+/* COL2ROW, ROW2COL*/
+#define DIODE_DIRECTION COL2ROW
+
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCE 5
+
+
+
```
Comment (fauxpark) [inline keyboards/free_willy/config.h:42]:
```suggestion
```

File: keyboards/free_willy/info.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
+{
+    "keyboard_name": "free willy",
+    "url": "https://www.squashkb.com/product/free-willy",
+    "maintainer": "tsquash",
+    "width": 10,
+    "height": 4,
```
Comment (fauxpark) [inline keyboards/free_willy/info.json:6]:
```suggestion
```

File: keyboards/free_willy/readme.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# Free Willy
+
+![Free Willy](https://imgur.com/a/RTpMSZ6)
```
Comment (fauxpark) [inline keyboards/free_willy/readme.md:3]:
```suggestion
![Free Willy](https://i.imgur.com/JBNV3Hk.png)
```
Needs to be a direct link.

File: keyboards/free_willy/readme.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+Make example for this keyboard (after setting up your build environment):
+
+    qmk compile -kb free_willy -km default
+    
+To reset the keyboard, jump the reset and ground pins located on the pro micro.
+
+Install example for this keyboard:
+
+    qmk flash -kb free_willy -km default
+
```
Comment (fauxpark) [inline keyboards/free_willy/readme.md:20]:
```suggestion
    make free_willy:default
    
To reset the keyboard, jump the reset and ground pins located on the pro micro.

Flashing example for this keyboard:

    make free_willy:default:flash
```

File: keyboards/free_willy/rules.mk [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+MOUSEKEY_ENABLE = no      # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no       # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+NKRO_ENABLE = no            # USB Nkey Rollover
+RGBLIGHT_ENABLE = no       # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+ENCODER_ENABLE = no
```
Comment (fauxpark) [inline keyboards/free_willy/rules.mk:15]:
```suggestion
# MCU name
MCU = atmega32u4

# Bootloader selection
BOOTLOADER = caterina

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**